src/contracts/middleware/ServiceManagerBase.sol

```diff
@@ -0,0 +1,145 @@
+    IDelegationManager public immutable delegationManager;
+    IStrategyManager public immutable strategyManager;
+    ISlasher public immutable slasher;
+
+    /// @notice when applied to a function, ensures that the function is only callable by the `registryCoordinator`.
+    modifier onlyRegistryCoordinator() {
+        require(
+            msg.sender == address(registryCoordinator),
+            "onlyRegistryCoordinator: not from registry coordinator"
+        );
+        _;
+    }
+
+    /// @notice when applied to a function, ensures that the function is only callable by the `registryCoordinator`.
+    /// or by StakeRegistry
+    modifier onlyRegistryCoordinatorOrStakeRegistry() {
+        require(
+            (msg.sender == address(registryCoordinator)) ||
+                (msg.sender ==
+                    address(
+                        IBLSRegistryCoordinatorWithIndices(
+                            address(registryCoordinator)
+                        ).stakeRegistry()
+                    )),
+            "onlyRegistryCoordinatorOrStakeRegistry: not from registry coordinator or stake registry"
+        );
+        _;
+    }
+
+    constructor(
+        IBLSRegistryCoordinatorWithIndices _registryCoordinator,
+        IStrategyManager _strategyManager,
+        IDelegationManager _delegationMananger,
+        ISlasher _slasher
+    ) BLSSignatureChecker(_registryCoordinator) {
+        strategyManager = _strategyManager;
+        delegationManager = _delegationMananger;
+        slasher = _slasher;
+        _disableInitializers();
+    }
+
+    function initialize(
+        IPauserRegistry _pauserRegistry,
+        address initialOwner
+    ) public initializer {
+        _initializePauser(_pauserRegistry, UNPAUSE_ALL);
+        _transferOwnership(initialOwner);
+    }
+
+    // PROXY CALLS TO EQUIVALENT SLASHER FUNCTIONS
+
+    /**
+     * @notice Called by the Registry in the event of a new registration, to forward a call to the Slasher
+     * @param operator The operator whose stake is being updated
+     */
+    function recordFirstStakeUpdate(
+        address operator
+    ) external onlyRegistryCoordinator {
+        slasher.recordFirstStakeUpdate(operator, latestServeUntilBlock());
+    }
+
+    /**
+     * @notice Called by the registryCoordinator, in order to forward a call to the Slasher, informing it of a stake update
+     * @param operator The operator whose stake is being updated
+     * @param updateBlock The block at which the update is being made
+     * @param prevElement The value of the previous element in the linked list of stake updates (generated offchain)
+     */
+    function recordStakeUpdate(
+        address operator,
+        uint32 updateBlock,
+        uint256 prevElement
+    ) external onlyRegistryCoordinatorOrStakeRegistry {
+        slasher.recordStakeUpdate(
+            operator,
+            updateBlock,
+            latestServeUntilBlock(),
+            prevElement
+        );
+    }
```

Comment:
we might want to mark these functions as 'virtual' (so they can be overridden) and/or add extensible "hooks" to them, but this seems like a fine start

Reply:
Made them virtual.